source/core/index.ts

```diff
@@ -0,0 +1,1472 @@
+			this._onResponse(response);
+		});
+
+		request.once('error', (error: Error) => {
+			if (error instanceof TimedOutTimeoutError) {
+				error = new TimeoutError(error, this.timings!, options);
+			} else {
+				error = new RequestError(error.message, error, options, this);
+			}
+
+			this._beforeError(error as RequestError);
+		});
+
+		this[kUnproxyEvents] = proxyEvents(request, this, proxiedRequestEvents);
+
+		this[kRequest] = request;
+
+		this.emit('uploadProgress', this.uploadProgress);
+
+		// Send body
+		const currentRequest = this.redirects.length === 0 ? this : request;
+		if (is.nodeStream(options.body)) {
+			options.body.pipe(currentRequest);
+			options.body.once('error', (error: NodeJS.ErrnoException) => {
+				this._beforeError(new UploadError(error, options, this));
+			});
+
+			options.body.once('end', () => {
+				delete options.body;
+			});
+		} else {
+			this._unlockWrite();
+
+			if (!is.undefined(options.body)) {
+				this._writeRequest(options.body, null as unknown as string, () => {});
+				currentRequest.end();
+
+				this._lockWrite();
+			} else if (this._cannotHaveBody || this._noPipe) {
+				currentRequest.end();
+
+				this._lockWrite();
+			}
+		}
+
+		this.emit('request', request);
+	}
+
+	async _makeRequest(): Promise<void> {
+		const {options} = this;
+		const {url, headers, request, agent, timeout} = options;
+
+		for (const key in headers) {
+			if (is.undefined(headers[key])) {
+				// eslint-disable-next-line @typescript-eslint/no-dynamic-delete
+				delete headers[key];
+			} else if (is.null_(headers[key])) {
+				throw new TypeError(`Use \`undefined\` instead of \`null\` to delete the \`${key}\` header`);
+			}
+		}
+
+		if (options.decompress && is.undefined(headers['accept-encoding'])) {
+			headers['accept-encoding'] = supportsBrotli ? 'gzip, deflate, br' : 'gzip, deflate';
+		}
+
+		// Set cookies
+		if (options.cookieJar) {
+			const cookieString: string = await options.cookieJar.getCookieString(options.url.toString());
+
+			if (is.nonEmptyString(cookieString)) {
+				options.headers.cookie = cookieString;
+			}
+		}
+
+		for (const hook of options.hooks.beforeRequest) {
+			// eslint-disable-next-line no-await-in-loop
+			const result = await hook(options);
+
+			if (!is.undefined(result)) {
+				// @ts-ignore Skip the type mismatch to support abstract responses
+				options.request = () => result;
+				break;
+			}
+		}
+
+		if (options.dnsCache && !('lookup' in options)) {
+			options.lookup = options.dnsCache.lookup;
+		}
+
+		// UNIX sockets
+		if (url.hostname === 'unix') {
+			const matches = /(?<socketPath>.+?):(?<path>.+)/.exec(`${url.pathname}${url.search}`);
+
+			if (matches?.groups) {
+				const {socketPath, path} = matches.groups;
+
+				Object.assign(options, {
+					socketPath,
+					path,
+					host: ''
+				});
+			}
+		}
+
+		const isHttps = url.protocol === 'https:';
+
+		let fallbackFn: HttpRequestFunction;
+		if (options.http2) {
+			fallbackFn = http2wrapper.auto;
```

Comment:
`options.http2` is `false` by default because big companies set their own agents, and many of them use HTTP1 Agents... Let's turn it on by default in Got 12.

Reply:
> Let's turn it on by default in Got 12.

Can you add a TODO comment?

Reply:
Done, put in `index.ts`